File: wrappers/dotnet/src/BbsSignatures/Models/IndexedMessage.cs
namespace BbsSignatures
{
    /// <summary>
    /// Represents a message and its index within a collection
    /// </summary>
    public struct IndexedMessage
    {
        /// <summary>
        /// The message
        /// </summary>
        public string Message;

        /// <summary>
        /// The message index
        /// </summary>
        public uint Index;


        /// <summary>
        /// Converts to string.
        /// </summary>
        /// <returns>
        /// A <see cref="System.String" /> that represents this instance.
        /// </returns>
        public override string ToString()
        {
            return $"{Index}: {Message}";
        }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/IndexedMessage.py
class IndexedMessage:

    __slots__ = ("message", "index")

    def __init__(self, message: str, index: int) -> None:
        """Represents a message and its index within a collection

        Args:
            message: The message
            index:  The message index
        """
        self.message = message
        self.index = index
